embed.go

```diff
@@ -25,26 +23,44 @@ var EmbeddedAppPath string
 //go:embed app.tar
 var embeddedApp []byte
 
+//go:embed app_checksum.txt
+var embeddedAppHash []byte
+
 func init() {
 	if len(embeddedApp) == 0 {
 		// No embedded app
 		return
 	}
 
-	h := md5.Sum(embeddedApp)
-	appPath := filepath.Join(os.TempDir(), "frankenphp_"+hex.EncodeToString(h[:]))
+	appPath := filepath.Join(os.TempDir(), "frankenphp_"+strings.TrimSuffix(string(embeddedAppHash[:]), "\n"))
 
-	if err := os.RemoveAll(appPath); err != nil {
-		panic(err)
-	}
-	if err := untar(appPath); err != nil {
-		os.RemoveAll(appPath)
-		panic(err)
+	if _, err := os.Stat(appPath); os.IsNotExist(err) {
+		mustUntar(appPath)
+	} else {
+		f, err := os.Open(appPath)
```

Comment:
I don't understand this block is necessary.

Reply:
This check is to ensure the directory is not empty. If the directory exists but is empty, then the archive should extracted

Reply:
But how could the directory be empty?

Reply:
It's probably just an edge case (E.G when the `untar` fails). I've removed the check, if someone runs into this then it can be added later